tests/shared.py

```diff
@@ -37,6 +39,53 @@ def first(c: Collection[T]) -> T:
     return next(iter(c))
 
 
+def load_graph_and_preserve_relative_ids(json_data, dummy_base="http://example.org/"):
+
+    rel_ids = set()
+
+    def collect_ids(obj):
+        if isinstance(obj, dict):
+            if "@id" in obj:
+                idv = obj["@id"]
+                if isinstance(idv, str) and (idv.startswith("./") or idv.startswith("../") or idv.startswith("#")):
```

Comment:
This also needs to catch ids like `data.csv` (relative URIs which may not have `./` or `/` at the beginning)

Also there should not be ids beginning with `../` - this is forbidden in RO-Crate